test/functional/test_framework/test_framework.py

```diff
@@ -519,11 +517,49 @@ def join_network(self):
         connect_nodes(self.nodes[1], 2)
         self.sync_all()
 
-    def sync_blocks(self, nodes=None, **kwargs):
-        sync_blocks(nodes or self.nodes, **kwargs)
+    def sync_blocks(self, nodes=None, *,  wait=1, timeout=60):
+        rpc_connections = nodes or self.nodes
+        """
+        Wait until everybody has the same tip.
 
-    def sync_mempools(self, nodes=None, **kwargs):
-        sync_mempools(nodes or self.nodes, **kwargs)
+        sync_blocks needs to be called with an rpc_connections set that has least
+        one node already synced to the latest, stable tip, otherwise there's a
+        chance it might return before all nodes are stably synced.
+        """
+        stop_time = time.time() + timeout
```

Comment:
The timeout should be scaled by `self.factor`, no? If yes, please add a commit for that.

Reply:
Wasn't aware of it as the methods weren't adapted in the *factor*-PR. Have used the original `self.options.factor` property to adjust timeouts for both methods.